token/services/db/sql/tokens.go

```diff
@@ -150,13 +128,20 @@ func (db *TokenDB) OwnersOf(txID string, index uint64) (*token.Token, []string,
 
 // Delete is called when spending a token
 func (db *TokenDB) Delete(txID string, index uint64, deletedBy string) error {
-	logger.Debugf("delete token [%s:%d:%s]", txID, index, deletedBy)
-	// We don't delete audit tokens, and we keep the 'ownership' relation.
-	now := time.Now().UTC()
-	query := fmt.Sprintf("UPDATE %s SET is_deleted = true, spent_by = $1, spent_at = $2 WHERE tx_id = $3 AND idx = $4;", db.table.Tokens)
-	logger.Debug(query, deletedBy, now, txID, index)
-	if _, err := db.db.Exec(query, deletedBy, now, txID, index); err != nil {
-		return errors.Wrapf(err, "error setting token to deleted [%s]", txID)
+	tx, err := db.NewTokenDBTransaction()
+	if err != nil {
+		return err
+	}
+	defer func() {
```

Comment:
Is it because if we have committed, the rollback won't take effect? If so, we can still spare the cycles of calling rollback by making a check whether `err` is set or something.